aws-api/src/main/java/com/amplifyframework/api/aws/GsonUtil.java

```diff
@@ -15,19 +15,34 @@
 
 package com.amplifyframework.api.aws;
 
+import com.amplifyframework.api.graphql.GraphQLResponse;
+import com.amplifyframework.core.model.scalar.AWSDate;
+import com.amplifyframework.core.model.scalar.AWSDateTime;
+import com.amplifyframework.core.model.scalar.AWSTime;
 import com.amplifyframework.util.Immutable;
 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
 import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
 
+import java.lang.reflect.Type;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 final class GsonUtil {
```

Comment:
All good thoughts!   

I considered the `GsonFactory` idea to allow an instance to be obtained from anywhere, but then realized that any implementation of `JsonDeserializer` should really just rely on the `JsonDeserializationContext` (which is passed in), instead of creating any new `Gson` instances.  With that in mind, we really only need the `Gson` instance in one place - in the `GsonGraphQLResponseFactory`, so I am now just creating the Gson instance inside the `GsonGraphQLResponseFactory` constructor, and having any sort of utility or factory isn't really necessary.

There wasn't a strong reason to keep the serializers and deserializers together, so I moved the Date `JsonSerializer` static classes into `GsonVariablesSerializer`, and then created a new `TemporalDeserializers` class to contain the Deserializers.